dynamics/tests/DGModelArray_test.cpp

```diff
@@ -0,0 +1,234 @@
+
+#include "include/ParametricMesh.hpp"
+
+namespace Nextsim {
+
+TEST_CASE("DGVector from ModelArray", "[DGModelArray]")
+{
+    static const int DG = 3;
+    const size_t nx = 32;
+    const size_t ny = 32;
+    const size_t mx = 100;
+    const size_t my = 100;
+
+    ModelArray::setDimensions(ModelArray::Type::DG, { nx, ny });
+    ModelArray::setNComponents(ModelArray::Type::DG, DG);
+
+    ModelArray source(ModelArray::Type::DG);
+    source.resize();
+    // Fill with data
+    for (size_t i = 0; i < nx; ++i) {
+        for (size_t j = 0; j < ny; j++) {
+            for (size_t c = 0; c < DG; ++c) {
+                source.components({i, j})[c] = c + my * (j + mx * (i));
+            }
+        }
+    }
+
+    CHECK(source.components({14, 12})[2] == (2 + my * (12 + mx * (14))));
+
+    // Create the ParametricMesh object
+    ParametricMesh smash;
+    smash.nx = nx;
+    smash.ny = ny;
+    smash.nnodes = nx * ny;
+    smash.nelements = nx * ny;
+    smash.vertices.resize(smash.nelements, Eigen::NoChange);
+    for (size_t i = 0; i < nx; ++i) {
+        for (size_t j = 0; j < ny; ++j) {
+            smash.vertices(i * ny + j, 0) = i;
+            smash.vertices(i * ny + j, 1) = j;
+        }
+    }
+    DGVector<DG> dest(smash);
+    DGModelArray::ma2dg<DG>(source, dest);
+
+    // Did it work?
+    size_t targetPoint = ModelArray::indexFromLocation(ModelArray::Type::DG, {14, 12});
+    REQUIRE(dest(targetPoint, 2) == source.components({14, 12})[2]);
+}
+
+TEST_CASE("ModelArray from DGVector", "[DGModelArray]")
+{
+    static const int DG = 6;
+    const size_t nx = 31;
+    const size_t ny = 33;
+    const size_t mx = 100;
+    const size_t my = 100;
+
+    // Create the ParametricMesh object
+    ParametricMesh smash;
+    smash.nx = nx;
+    smash.ny = ny;
+    smash.nnodes = nx * ny;
+    smash.nelements = nx * ny;
+    smash.vertices.resize(smash.nelements, Eigen::NoChange);
+    for (size_t i = 0; i < nx; ++i) {
+        for (size_t j = 0; j < ny; ++j) {
+            smash.vertices(i * ny + j, 0) = i;
+            smash.vertices(i * ny + j, 1) = j;
+        }
+    }
+    DGVector<DG> source(smash);
+    // Let the H arrays do the index calculation
+    ModelArray::setDimensions(ModelArray::Type::H, { nx, ny });
+    for (size_t i = 0; i < nx; ++i) {
+        for (size_t j = 0; j < ny; j++) {
+            size_t index = ModelArray::indexFromLocation(ModelArray::Type::H, {i, j});
+            for (size_t c = 0; c < DG; ++c) {
+                source(index, c) = c + my * (j + mx * (i));
+            }
+        }
+    }
+    size_t targetPoint = ModelArray::indexFromLocation(ModelArray::Type::H, {14, 12});
+    CHECK(source(targetPoint, 4) == 4 + my * (12 + mx *(14)));
+
+    ModelArray::setDimensions(ModelArray::Type::DG, { nx, ny });
+    ModelArray::setNComponents(ModelArray::Type::DG, DG);
+
+    ModelArray dest(ModelArray::Type::DG);
+    dest.resize();
+    DGModelArray::dg2ma(source, dest);
+
+    // Did it work?
```

Comment:
Comment: General point (not picking on this line specifically). We could perhaps do with expanded comments above each test case to define a test objective and talk about which files/classes/methods are being targeted. But comments above each require (like in the test below this) should be encouraged). Should improve debug speed, increase our bus number, make it easier to deprecate/enhance features, etc.

Reply:
(Also, I'm being very careful when crossing the road.)